src/AutoRest.CSharp.V3/Input/Source/ModelTypeMapping.cs

```diff
@@ -2,31 +2,61 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
 using System.Linq;
+using Azure.Core;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 
 namespace AutoRest.CSharp.V3.Input.Source
 {
-    public class ModelTypeMapping
+    public class ModelTypeMapping: TypeMapping
     {
-        public string SchemaName { get; }
-        public INamedTypeSymbol ExistingType { get; }
-
         public IMethodSymbol? DefaultConstructor { get; }
         public SourceMemberMapping[] PropertyMappings { get; }
 
-        public ModelTypeMapping(string schemaName, INamedTypeSymbol existingType, SourceMemberMapping[] propertyMappings)
+        public ModelTypeMapping(INamedTypeSymbol memberAttribute, string originalName, INamedTypeSymbol existingType): base(originalName, existingType)
         {
-            SchemaName = schemaName;
-            ExistingType = existingType;
-            PropertyMappings = propertyMappings;
+            List<SourceMemberMapping> memberMappings = new List<SourceMemberMapping>();
+            foreach (var member in GetMembers(existingType))
```

Comment:
Nit: not sure how close you're following convention in azure-sdk-for-net, but you have flipped when it's okay and not okay to use `var` here (okay for `new`; not okay otherwise).